tests/ttnn/unit_tests/operations/eltwise/test_float32.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+import torch
+
+import ttnn
+from models.utility_functions import is_grayskull
+
+from tests.ttnn.utils_for_testing import assert_with_pcc
+from models.utility_functions import torch_random
+
+from loguru import logger
+
+
+# https://github.com/tenstorrent/tt-metal/issues/14862
+def test_unary_float32(device):
+    torch.manual_seed(0)
+
+    torch.set_printoptions(precision=10)
+
+    torch_input_tensor = torch.tensor([[0.00001]], dtype=torch.float32)
+    torch_output_tensor = -torch_input_tensor
+
+    ttnn_input_tensor = ttnn.from_torch(torch_input_tensor, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
+    import os
+
+    print(f"Debugging PID: {os.getpid()}")
+    ttnn_output_tensor = ttnn.neg(ttnn_input_tensor)
+
+    print(torch_input_tensor, ttnn.to_torch(ttnn_input_tensor))
+    print(torch_output_tensor, ttnn.to_torch(ttnn_output_tensor))
+
+    output_tensor = ttnn.to_torch(ttnn_output_tensor)
+
+    assert True == torch.allclose(torch_output_tensor, output_tensor, atol=1e-8, rtol=1e-5, equal_nan=False)
+
```

Comment:
@KalaivaniMCW , see if we can get the assert with atol=1e-8 to be atol=1e-10 on your PR [#14899](https://github.com/tenstorrent/tt-metal/pull/14899)